app/actions/handlers.py

```diff
@@ -0,0 +1,64 @@
+import json
+import logging
+
+from app.actions.configurations import AuthenticateConfig, PullObservationsConfig
+from app.services.state import IntegrationStateManager
+
+
+logger = logging.getLogger(__name__)
+
+
+state_manager = IntegrationStateManager()
+
+
+async def action_auth(integration, action_config: AuthenticateConfig):
+    logger.info(f"Executing auth action with integration {integration} and action_config {action_config}...")
+    if action_config.password == "valid_password" and action_config.username == "valid_username":
+        return {"valid_credentials": True}
+    else:
+        return {"valid_credentials": False}
+
+
+async def action_fetch_samples(integration, action_config: PullObservationsConfig):
+    logger.info(f"Executing fetch_samples action with integration {integration} and action_config {action_config}...")
+    vehicles = {
```

Comment:
You might consider using a name other than `vehicles` here -- since this is a demo repository, it might be confusing if we have a list of things called vehicles that is really a list of geo-locations or observations.